facade/service.go

```diff
@@ -1230,6 +1231,31 @@ func (f *Facade) clearEmergencyStopFlag(ctx datastore.Context, tenantID, service
 	return cleared, nil
 }
 
+// countServices will count how many unique services in the given list (including children) satisfy the check function
+func (f *Facade) countServices(ctx datastore.Context, serviceIDs []string, check func(*service.Service) bool) (int, error) {
+	alreadyChecked := make(map[string]bool)
```

Comment:
Instead of a bool, this could be a map[string]struct{}, and you would be checking the presence of key in the struct instead of checking the bool value.  The advantage of this is that the struct{} type takes up 0 bytes, while the bool type takes > 0 bytes.
```go
checked := make(map[string]struct{})
a, ok := checked[name]
//ok represents the presence of the element in the map
```